.github/workflows/rust-ci-cd.yml

```
@@ -4,57 +4,102 @@ on:
+      - uses: actions/checkout@v4
+
+      - name: ▶ Allow Workflow Run
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          echo "This workflow was triggered manually."
+          echo "run_ci_cd=true" >> $GITHUB_ENV
+
+      - name: 🔎 Check if CI/CD is needed
+        if: ${{ github.event_name == 'pull_request' || github.event_name == 'push' }}
+        id: check-ci-cd
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:4:28: Double quote to prevent globbing and word splitting

(shellcheck)